ntoskrnl/inbv/inbv.c

```diff
@@ -932,18 +948,67 @@ DisplayBootBitmap(IN BOOLEAN TextMode)
     }
 
 #ifdef INBV_ROTBAR_IMPLEMENTED
+    /* Hide simple progress bar if not used */
+    if (RotBarSelection != RB_PROGRESS_BAR)
+        ShowProgressBar = FALSE;
+
     /* Do we have a system thread? */
     if (RotBarThreadActive)
     {
         /* We do, initialize the progress bar */
         InbvAcquireLock();
         RotBarSelection = TempRotBarSelection;
-        // InbvRotBarInit();
+        InbvRotBarInit();
         InbvReleaseLock();
     }
 #endif
 }
 
+VOID
+NTAPI
+INIT_FUNCTION
+InbvRotBarInit(VOID)
+{
+    ULONG X, Y, Iteration, Index, Total = 18;
+    UCHAR Buffer[128];
+#ifndef INBV_ROTBAR_IMPLEMENTED
+    return;
+#endif
+    if (RotBarSelection != RB_SQUARE_CELLS)
+        return;
+
+    /* This function needs to be in system thread, but it's not yet */
```

Comment:
Do you mean "needs to be called by system thread" ?

Reply:
This part of code should be in `while ()` loop instead of `for`, and run in context of a separate thread.

For now it's executed sequentially, so the loop is limited to 6 rotations (for demonstration).